ios/Classes/AblyInstanceStore.m

```diff
@@ -65,20 +65,21 @@ -(ARTPaginatedResult *) getPaginatedResult:(NSNumber *const) handle {
     return _paginatedResults[handle];
 }
 
-// Set device token on all existing clients
--(void) didRegisterForRemoteNotificationsWithDeviceToken:(NSData *const) deviceToken {
-    _didRegisterForRemoteNotificationsWithDeviceToken_deviceToken = deviceToken;
-    
-    for (id restHandle in _restInstances) {
-        ARTRest *const rest = _restInstances[restHandle];
-        [ARTPush didRegisterForRemoteNotificationsWithDeviceToken:deviceToken rest:rest];
-    }
-    for (id realtimeHandle in _realtimeInstances) {
-        ARTRealtime *const realtime = _realtimeInstances[realtimeHandle];
+// Set device token for the first created realtime object, device of which is available for others, because it's static
+-(void)didRegisterForRemoteNotificationsWithDeviceToken:(NSData *const)deviceToken {
+    ARTRealtime *const realtime = _realtimeInstances.allValues.firstObject;
```

Comment:
do we actually need this code? In what circumstances we have to call `didRegisterForRemoteNotificationsWithDeviceToken` or `didFailToRegisterForRemoteNotificationsWithError`? Shouldn't this be a part of state machine logic?

Reply:
Actually, no, we don't. But I've tried to simplify what is already here. I think you are right and we should remove all the APNs stuff from flutter.

Reply:
I've changed how it works - https://github.com/ably/ably-flutter/pull/534/commits/18d12079312fec8bf63e8c869ba7d3b8aa24e67d

Reply:
Now it stores the realtime/rest handle through which the token was requested and stores the token using the same realtime/rest instance. I think we should leave this code since removing it will break user's apps.